### PR TITLE
fix: update pin screens header

### PR DESCRIPTION
**Pull request by jleach**
# Summary of Changes

- Add the header to both the PIN create and enter screens;
- Cleanup a testID and update unit tests acordingly.

# Related Issues

Please reference here any issue #'s that are relevant to this PR, or simply enter "N/A" if this PR does not relate to any existing issues.

# Pull Request Checklist

Tick all boxes below to demonstrate that you have completed the respective task. If the item does not apply to your this PR **check it anyway** to make it apparent that there's nothing to do.

- [x] All commits contain a DCO `Signed-off-by` line (we use the [DCO GitHub app](https://github.com/apps/dco) to enforce this);
- [x] Updated LICENSE-3RD-PARTY.md for any added dependencies or vendored components;
- [x] Updated documentation as needed for changed code and new or modified features;
- [x] Added sufficient [tests](../__tests__/) so that overall code coverage is not reduced.

If you have _any_ questions to _any_ of the points above, just **submit and ask**! This checklist is here to _help_ you, not to deter you from contributing!

Pro Tip 🤓

- Read our [contribution guide](../CONTRIBUTING.md) at least once; it will save you a few review cycles!
- Your PR will likely not be reviewed until all the above boxes are checked and all automated tests have passed.

_PR template adapted from the Python attrs project._

### CODE REVIEW

File: core/App/navigators/RootStack.tsx [hunk 1]
```diff
@@ -104,7 +104,14 @@ const RootStack: React.FC = () => {
     return (
       <Stack.Navigator initialRouteName={Screens.Splash} screenOptions={{ ...defaultStackOptions, headerShown: false }}>
         <Stack.Screen name={Screens.Splash} component={splash} />
-        <Stack.Screen name={Screens.EnterPin}>
+        <Stack.Screen
+          name={Screens.EnterPin}
+          options={() => ({
+            headerShown: true,
+            headerLeft: () => false,
+            rightLeft: () => false,
```
Comment (jcdrouin21) [inline core/App/navigators/RootStack.tsx]: Can we make sure that we can translate the header with: `title: t(..)`

File: core/App/navigators/RootStack.tsx [hunk 2]
```diff
@@ -169,7 +176,14 @@ const RootStack: React.FC = () => {
           })}
           component={terms}
         />
-        <Stack.Screen name={Screens.CreatePin}>
+        <Stack.Screen
+          name={Screens.CreatePin}
+          options={() => ({
+            headerShown: true,
+            headerLeft: () => false,
+            rightLeft: () => false,
```
Comment (jcdrouin21) [inline core/App/navigators/RootStack.tsx]: Same here.

File: core/App/navigators/RootStack.tsx (hunk 1 above)
Comment (jleach) [reply core/App/navigators/RootStack.tsx]: Fixed

File: core/App/navigators/RootStack.tsx (hunk 2 above)
Comment (jleach) [reply core/App/navigators/RootStack.tsx]: Fixed

**Review by amanji — APPROVED** [review]

**Review by amanji — APPROVED** [review]

**Pull request merged**